diff --git a/dist/index.js b/dist/index.js
index 1351363e79aa6b3506454cbdd63ea03cd65c56af..16fe2c96b8d9ee6f767d52728520d5e8ffb465e5 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -4586,12 +4586,12 @@ var _Provider = class {
   static ensureClientVersionIsSupported(nodeInfo) {
     const { isMajorSupported, isMinorSupported, supportedVersion } = (0, import_versions.checkFuelCoreVersionCompatibility)(nodeInfo.nodeVersion);
     if (!isMajorSupported || !isMinorSupported) {
-      console.warn(
-        `The Fuel Node that you are trying to connect to is using fuel-core version ${nodeInfo.nodeVersion},
-which is not supported by the version of the TS SDK that you are using.
-Things may not work as expected.
-Supported fuel-core version: ${supportedVersion}.`
-      );
+//       console.warn(
+//         `The Fuel Node that you are trying to connect to is using fuel-core version ${nodeInfo.nodeVersion},
+// which is not supported by the version of the TS SDK that you are using.
+// Things may not work as expected.
+// Supported fuel-core version: ${supportedVersion}.`
+//       );
     }
   }
   /**
diff --git a/dist/index.mjs b/dist/index.mjs
index 2843048856717ce34c5588a167b5c7f823ba7433..1c8c63b79951bed755f5130003bbb142c0d11763 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -4420,12 +4420,12 @@ var _Provider = class {
   static ensureClientVersionIsSupported(nodeInfo) {
     const { isMajorSupported, isMinorSupported, supportedVersion } = checkFuelCoreVersionCompatibility(nodeInfo.nodeVersion);
     if (!isMajorSupported || !isMinorSupported) {
-      console.warn(
-        `The Fuel Node that you are trying to connect to is using fuel-core version ${nodeInfo.nodeVersion},
-which is not supported by the version of the TS SDK that you are using.
-Things may not work as expected.
-Supported fuel-core version: ${supportedVersion}.`
-      );
+//       console.warn(
+//         `The Fuel Node that you are trying to connect to is using fuel-core version ${nodeInfo.nodeVersion},
+// which is not supported by the version of the TS SDK that you are using.
+// Things may not work as expected.
+// Supported fuel-core version: ${supportedVersion}.`
+//       );
     }
   }
   /**
